src/pipx/backend.py

```diff
@@ -0,0 +1,24 @@
+import logging
+import os
+import shutil
+
+DEFAULT_BACKEND = os.getenv("PIPX_DEFAULT_BACKEND", "venv")
+DEFAULT_INSTALLER = os.getenv("PIPX_DEFAULT_INSTALLER", "pip")
+SUPPORTED_VENV_BACKENDS = ("uv", "venv", "virtualenv")
+SUPPORTED_INSTALLERS = ("uv", "pip")
+
+logger = logging.getLogger(__name__)
+
+
+def path_to_exec(executable: str, installer: bool = False) -> str:
+    if executable in ("venv", "pip"):
+        return executable
+    path = shutil.which(executable)
+    if path:
+        return path
+    elif installer:
+        logger.warning(f"'{executable}' not found on PATH. Falling back to 'pip'.")
+        return ""
+    else:
+        logger.warning(f"'{executable}' not found on PATH. Falling back to 'venv'.")
+        return ""
```

Comment:
I find the `installer` argument confusing here. Everywhere else, `installer` is a `str` of the installer, not a boolean. IMO having an `ExecType` enum here with `INSTALLER` and `BACKEND` would make this function much clearer.

Reply:
Or maybe we can simply rename `installer` argument to `is_installer`?